tests/mock_loginsight_server/hosts.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+import requests_mock
+import json
+import logging
+
+from .utils import requiresauthentication
+
+mockserverlogger = logging.getLogger("LogInsightMockAdapter")
+
+
+class MockedHoststMixin(requests_mock.Adapter):
+
+    def __init__(self, **kwargs):
+        super(MockedHoststMixin, self).__init__(**kwargs)
+
+        self.known_sources = [
+            {'lastReceived': 1505247698798, 'sourcePath': '192.168.0.66', 'sortOrder': '0'},
+        ]
+
+        self.known_hosts = [
```

Comment:
The mock returns a small number of entries, shorter than the pagination size. That means the pagination-code is not exercised by the mock. It should be. Generate a large number (50k?) hosts deterministically here.

Reply:
I wrote a function that generates 300 fake hosts.